src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidRunCommand.cs

```diff
@@ -0,0 +1,267 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.DotNet.XHarness.Android;
+using Microsoft.DotNet.XHarness.Android.Execution;
+using Microsoft.DotNet.XHarness.CLI.CommandArguments.Android;
+using Microsoft.DotNet.XHarness.Common.CLI;
+using Microsoft.DotNet.XHarness.Common.CLI.CommandArguments;
+using Microsoft.DotNet.XHarness.Common.CLI.Commands;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.XHarness.CLI.Commands.Android
+{
+    internal class AndroidRunCommand : XHarnessCommand
+    {
+        // nunit2 one should go away eventually
+        private static readonly string[] s_xmlOutputVariableNames = { "nunit2-results-path", "test-results-path" };
+        private const string TestRunSummaryVariableName = "test-execution-summary";
+        private const string ShortMessageVariableName = "shortMsg";
+        private const string ReturnCodeVariableName = "return-code";
+        private const string ProcessCrashedShortMessage = "Process crashed";
+
+        private readonly AndroidRunCommandArguments _arguments = new AndroidRunCommandArguments();
+
+        protected override XHarnessCommandArguments Arguments => _arguments;
+
+        protected override string CommandUsage { get; } = "android run [OPTIONS]";
+
+        private const string CommandHelp = "Run tests on already installed .apk on an Android device";
+        protected override string CommandDescription { get; } = @$"
+{CommandHelp}
+
+APKs can communicate status back to XHarness using the results bundle:
+
+Required:
```

Comment:
```Optional:
Test results Paths:
nunit2-results-path
test-results-path - If specified, this file will be copied off the device after execution (used for external reporting)
Reporting:
test-execution-summary,shortMsg - If specified, this will be printed to the console directly after execution (useful for printing summaries)

Arguments:


      --app, -a=VALUE        Path to already-packaged app
      --output-directory, -o=VALUE
                             Directory in which the resulting package will be
                               outputted
      --targets, -t=VALUE    Comma-delimited list of targets to test for
      --timeout=VALUE        TimeSpan or number of seconds to wait for
                               instrumentation to complete
      --method, -m=VALUE     Method to be ran in the test application. When
                               this parameter is used only the tests that have
                               been provided by the '--method' and '--class'
                               arguments will be ran. All other test will be
                               ignored. Can be used more than once.
      --class, -c=VALUE      Test class to be ran in the test application. When
                               this parameter is used only the tests that have
                               been provided by the '--method' and '--class'
                               arguments will be ran. All other test will be
                               ignored. Can be used more than once.
      --device-out-folder, --dev-out=VALUE
                             If specified, copy this folder recursively off the
                               device to the path specified by the output
                               directory
      --instrumentation, -i[=VALUE]
                             If specified, attempt to run instrumentation with
                               this name instead of the default for the
                               supplied APK.
      --expected-exit-code=VALUE
                             If specified, sets the expected exit code for a
                               successful instrumentation run.
      --package-name, -p=VALUE
                             Package name contained within the supplied APK
      --arg=VALUE            Argument to pass to the instrumentation, in form
                               key=value
      --verbosity, -v[=VALUE]
                             Verbosity level - defaults to 'Information' if not
                               specified. If passed without value, 'Debug' is
                               assumed (highest)
      --help, -h

Reply:
I am still somehow confused about the top paragraph and what message it's trying to deliver. Maybe users of XHarness will be too?